### PR TITLE
Removing console_bridge

### PR DESCRIPTION
**Pull request by CursedRock17**
This pull request directly addresses issue #570 which is meant to remove the `console_bridge` dependencies from tf2 and opt for a ROS 2 solution of logging. I went with the basic forms of `RCUTILS_LOG_WARN/ERROR` as it seems to be most sensible. Though, this could be swapped to `RCLCPP_LOG_*` or more advanced forms like `RCUTILS_LOG_WARN_NAMED` depending on the extent of the information we want the user to have.

### CODE REVIEW

**Review by ahcorde — CHANGES_REQUESTED** [review]
There are 3 new warnings, do you mind to fix them?

https://build.ros2.org/job/Rpr__geometry2__ubuntu_jammy_amd64/263/clang-tidy/new/

**Comment by CursedRock17** [comment]
Yup, they're all fixed now, it seems CI errors are now related to flake8 tests in other parts of the repository.

File: tf2/src/buffer_core.cpp [hunk 1]
```diff
@@ -678,7 +677,7 @@ void BufferCore::lookupTransformImpl(
       case tf2::TF2Error::TF2_LOOKUP_ERROR:
         throw LookupException(error_string);
       default:
-        CONSOLE_BRIDGE_logError("Unknown error code: %d", retval);
+        RCUTILS_LOG_ERROR("Unknown error code: %d", retval);
```
Comment (ahcorde) [inline tf2/src/buffer_core.cpp]:
```suggestion
        RCUTILS_LOG_ERROR("Unknown error code: %u", static_cast<std::uint8_t>(retval));
```

**Review by ahcorde — CHANGES_REQUESTED** [review]
warning are still there https://build.ros2.org/job/Rpr__geometry2__ubuntu_jammy_amd64/264/clang-tidy/new/

TF2Error is a enum of `std::uint8_t`
```cpp
enum class TF2Error : std::uint8_t
```

Take a look to my suggestion

**Comment by ahcorde** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/20406/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/14816/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/21156/)

**Review by ahcorde — APPROVED** [review]

**Pull request merged**